### ISSUE TITLE
[java] InefficientStringBuffering false positive

### ISSUE THREAD

**Issue by vbrandl** [pmd/pmd#1631]
**Affects PMD Version:** 6.11.0

**Rule:** InefficientStringBuffering

**Description:** When calculating the initial capacity of a string buffer, PMD falsely interprets this as a string concatenation.

**Code Sample demonstrating the issue:**

```java
public final class Test {
  public String foo(final String x, final String y, final int z) {
    final StringBuilder sb = new StringBuilder(x.length() + (y.length() / z));
    sb.append(bar());
    return sb.toString();
  }

  public String bar() {
    throw new RuntimeException();
  }
}
```

**Running PMD through:** *Gradle*

**Comment by jsotuyod** [comment]
This is no longer reproducible under PMD 7
